geode-core/src/main/java/org/apache/geode/internal/cache/xmlcache/CacheCreation.java

```diff
@@ -521,12 +521,12 @@ void create(InternalCache cache)
 
     cache.initializePdxRegistry();
 
-    for (DiskStore diskStore : diskStores.values()) {
+    diskStores.values().parallelStream().forEach(diskStore -> {
```

Comment:
Would it be possible to put a bound on the number of threads used here? In cases with large numbers of disk stores this might cause resource issues if unbounded, but maybe we never have so many disk stores that it will matter.

Reply:
In Geode use case, it won't be large number of diskstore. The worst case is less than region number.

Reply:
Parallel stream by default uses [`ForkJoinPool`](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ForkJoinPool.html) with parallelism equals the number of processors.